flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -238,18 +241,19 @@ private InputGate createInputGate(TaskManagerLocation senderLocation) throws Exc
 
 	private InputGateDeploymentDescriptor createInputGateDeploymentDescriptor(
 			TaskManagerLocation senderLocation,
-			int consumedSubpartitionIndex,
+			int gateIndex,
 			ResourceID localLocation) {
 
-		final ShuffleDescriptor[] channelDescriptors = Arrays.stream(partitionIds)
-			.map(partitionId ->
-				createShuffleDescriptor(localMode, partitionId, localLocation, senderLocation, consumedSubpartitionIndex))
-			.toArray(ShuffleDescriptor[]::new);
+		final ShuffleDescriptor[] channelDescriptors = new ShuffleDescriptor[channels];
+		for (int channelIndex = 0; channelIndex < channels; ++channelIndex) {
+			channelDescriptors[channelIndex] = createShuffleDescriptor(
+				localMode, partitionIds[gateIndex], localLocation, senderLocation, gateIndex);
+		}
 
 		return new InputGateDeploymentDescriptor(
 			dataSetID,
 			ResultPartitionType.PIPELINED_BOUNDED,
-			consumedSubpartitionIndex,
+			0,
```

Comment:
This is always zero, as it's being ignored by `Test***InputChannel` classes, right?

If so, maybe add a comment explaining that?